index.js

```diff
@@ -20,35 +22,293 @@ class J5sExecutor extends Executor {
+            action: 'get',
+            params: [{ name: jobName }]
+        }).then((data) => {
+            if (data && data.lastCompletedBuild && data.lastCompletedBuild.number) {
+                return null;
+            }
+
+            // delay between retry attempts
+            return new Promise((resolve) => {
+                setTimeout(() => resolve(), this.cleanupWatchInterval * 1000);
+            }).then(() =>
+                this._jenkinsJobWaitStop(jobName, timeConsumed + this.cleanupWatchInterval)
+            );
+        });
+    }
+
+    /**
+     * Jenkins job config xml
+     * @method _loadJobXml
+     * @param  {Object}   config        A configuration object psssed to _start
+     * @return {Promise}
+     */
+    _loadJobXml(config) {
+        const { buildScript, cleanupScript } = this._taskScript(config);
+
+        return new Promise((resolve, reject) => {
+            const configPath = path.resolve(__dirname, './config/job.xml');
+
+            fs.readFile(configPath, 'utf-8', (err, fileContents) => {
+                if (err) {
+                    return reject(err);
+                }
+
+                return resolve(fileContents);
+            });
+        }).then(xml => (
+            _.template(xml)({
+                nodeLabel: xmlescape(this.nodeLabel),
+                buildScript: xmlescape(buildScript),
+                cleanupScript: xmlescape(cleanupScript)
+            })
+        ));
+    }
+
+    /**
+     * Script for job
+     * @method _taskScript
+     * @param  {Object}   config        A configuration object psssed to _start
+     * @return {Object}
+     */
+    _taskScript(config) {
+        if (this.buildScript && this.cleanupScript) {
+            return {
+                buildScript: this.buildScript,
+                cleanupScript: this.cleanupScript
+            };
+        }
+
+        return this._dockerTaskScript(config);
+    }
+
+    /**
+     * Script for job using docker
+     * @method _dockerTaskScript
+     * @param  {Object}   config        A configuration object psssed to _start
+     * @return {Object}
+     */
+    _dockerTaskScript(config) {
+        const pullLauncherImage = shellescape([
+            this.dockerCommand, 'pull', `screwdrivercd/launcher:${this.launchVersion}`
+        ]);
+        const pullBuildImage = shellescape([
+            this.dockerCommand, 'pull', config.container
+        ]);
+        const launcherContainerName = `${this.prefix}${config.buildId}-init`;
+        const createLauncherContainer = shellescape([
+            this.dockerCommand, 'run',
+            '--name', launcherContainerName,
+            '--label', `sdbuild=${this.prefix}${config.buildId}`,
+            '--entrypoint', '/bin/true',
+            `screwdrivercd/launcher:${this.launchVersion}`
+        ]);
+        const buildContainerName = `${this.prefix}${config.buildId}-build`;
+        const createBuildContainer = shellescape([
+            this.dockerCommand, 'run',
```

Comment:
Is it possible to make this block into a template and replace the values? It looks hard to read here.